### PR TITLE
[PEFT] Support low_cpu_mem_usage option for PEFT loading adapters

### PR DESCRIPTION
**Pull request by BenjaminBossan**
# What does this PR do?

PEFT added support for `low_cpu_mem_usage=True` when loading adapters in https://github.com/huggingface/peft/pull/1961. This feature is now available when installing PEFT v0.13.0. With this PR, this option is also supported when loading PEFT adapters directly into transformers models.

Additionally, with this PR, https://github.com/huggingface/diffusers/pull/9510 will be unblocked, which implements this option in diffusers.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sayakpaul** [comment]
@ArthurZucker could you give this a look?

File: src/transformers/integrations/peft.py [hunk 1]
```diff
@@ -129,12 +133,27 @@ def load_adapter(
...
+                Reduce memory usage while loading the PEFT adapter. This should also speed up the loading process.
+                Requires PEFT version 0.13.0 or higher.
             adapter_kwargs (`Dict[str, Any]`, *optional*):
                 Additional keyword arguments passed along to the `from_pretrained` method of the adapter config and
                 `find_adapter_config_file` method.
         """
         check_peft_version(min_version=MIN_PEFT_VERSION)
 
+        # peft only supports low_cpu_mem_usage starting from v0.13.0
+        peft_load_kwargs = {}
```
Comment (ArthurZucker) [inline src/transformers/integrations/peft.py:146]: LGTM, not sure we need all the kwargs as you are not taking them from the kwargs of this function 😉

**Review by ArthurZucker — APPROVED** [review]

File: src/transformers/integrations/peft.py (hunk 1 above)
Comment (BenjaminBossan) [reply src/transformers/integrations/peft.py:146]: Not sure if I get you, do you mean this should be merged with `adapter_kwargs`? That wouldn't work, as these are `kwargs` used for a different purpose. 

Or that using a dict for a single argument is overkill?

Comment (ArthurZucker) [reply src/transformers/integrations/peft.py:146]: > Or that using a dict for a single argument is overkill?
this 🤗

Comment (BenjaminBossan) [reply src/transformers/integrations/peft.py:146]: I found this more readable and extensible than having:

```python
if low_cpu_mem_usage:
    inject_adapter_in_model(peft_config, self, adapter_name, low_cpu_mem_usage=low_cpu_mem_usage)
else:
    inject_adapter_in_model(peft_config, self, adapter_name)
```

(same with the `set_peft_model_state_dict` call)

If you want me to change it to this instead or have an alternative idea, let me know and I'll change it.

Comment (ArthurZucker) [reply src/transformers/integrations/peft.py:146]: no worries

**Comment by ArthurZucker** [comment]
not sure what's going on with the tests, rebasing should help !

**Comment by BenjaminBossan** [comment]
@ArthurZucker I still get `fetch_tests - Failed` from `setup_and_quality`. Any idea what could be the cause? I ran `ruff` on the touched files.

**Comment by ArthurZucker** [comment]
I had to force push as the rebasing you did went a bit wrong! Should be good now! 🤗

**Comment by ArthurZucker** [comment]
Nope, I was just waiting for the cis!

**Pull request merged**